tools/scalar-schema/src/scalar_schema/jdbc.clj

```diff
@@ -0,0 +1,298 @@
+
+(defn- insert-metadata
+  [db-spec rdb-engine prefix schema table partition-key clustering-key columns secondary-index]
+  (doall (map #(let [column (first (first %))
+                     data-type (string/upper-case (second (first %)))
+                     key-type (get-key-type column partition-key clustering-key)
+                     key-order (if (key? column clustering-key) "ASC" nil)
+                     secondary-indexed (get-secondary-indexed rdb-engine column secondary-index)
+                     index-order (if (key? column secondary-index) "ASC" nil)
+                     ordinal-position (second %)]
+                 (execute! db-spec (make-insert-metadata-statement rdb-engine prefix schema table
+                                                                   column data-type key-type
+                                                                   key-order secondary-indexed
+                                                                   index-order ordinal-position)))
+              (zipseq columns (iterate inc 1)))))
+
+(defn- make-drop-table-statement
+  [rdb-engine schema table]
+  (str "DROP TABLE " (get-table-name rdb-engine schema table)))
+
+(defn- drop-table
+  [db-spec rdb-engine schema table]
+  (execute! db-spec (make-drop-table-statement rdb-engine schema table)))
+
+(defn- make-delete-metadata-statement
+  [rdb-engine prefix schema table]
+  (let [full_table_name (get-table-name schema table)]
+    (str "DELETE FROM "
+         (get-metadata-table-name rdb-engine prefix)
+         " WHERE " (enclose rdb-engine "full_table_name") " = "
+         "'" full_table_name "'")))
+
+(defn- delete-metadata
+  [db-spec rdb-engine prefix schema table]
+  (execute! db-spec (make-delete-metadata-statement rdb-engine prefix schema table)))
+
+(defn- reorder-columns
+  [columns partition-key clustering-key]
+  (concat (map #(vector % (get columns %)) partition-key)
+          (map #(vector % (get columns %)) clustering-key)
+          (filter #(not (key? (first %) (concat partition-key clustering-key)))
+                  (apply list columns))))
+
+(defn make-jdbc-operator
+  [{:keys [jdbc-url user password prefix]}]
+  (let [db-spec (get-db-spec jdbc-url user password)
+        rdb-engine (get-rdb-engine jdbc-url)]
+    (create-metadata-schema db-spec rdb-engine prefix)
+    (create-metadata-table db-spec rdb-engine prefix)
+    (reify proto/IOperator
+      (create-table [_ {:keys [database table partition-key clustering-key columns
+                               secondary-index]} _]
+        (let [reordered-columns (reorder-columns columns partition-key clustering-key)
+              table-name (get-table-name database table)]
+          (try
+            (create-schema db-spec rdb-engine database)
+            (create-table db-spec rdb-engine database table partition-key clustering-key
+                          reordered-columns secondary-index)
```

Comment:
I think it is better to give a map to each function in Clojure instead of lots of arguments.
In this case, the following looks simpler.
```clojure
(create-table db-spec rdb-engine schema)
```
If the `opts` has an executor and `rdb-engine`,
```clojure
(create-table schema opts)
```